### PR TITLE
Migrate to ESM and upgrade dependencies

### PR DESCRIPTION
**Pull request by sandhose**
I would like to explore exporting icons as react components directly.
I noticed we were already exporting some JS, but with ESM syntax, and without `type: module` in the package.json.
So, I switched the whole package to ESM, which of course broke some stuff in the build process.

I fixed the build process, and then upgraded all the dev dependencies to the latest version.

Because of the ongoing reformat, I've rebased this on top of #87

### CODE REVIEW

File: assets/ios/swift/CompoundColorTokens.swift
Comment (sandhose) [inline assets/ios/swift/CompoundColorTokens.swift:1]: Only change here are trailing spaces at the end of lines

File: assets/ios/swift/CompoundIcons.swift
Comment (sandhose) [inline assets/ios/swift/CompoundIcons.swift:1]: Same here, trailing spaces

File: assets/web/css/cpd-common-base.css [hunk 1]
```diff
@@ -35,7 +35,7 @@
   --cpd-space-11x: 44px;
   --cpd-space-10x: 40px;
   --cpd-space-6x: 24px;
-  --cpd-space-0x: 0px;
+  --cpd-space-0x: 0;
```
Comment (sandhose) [inline assets/web/css/cpd-common-base.css:38]: This shouldn't break anything

File: assets/web/css/cpd-common-base.css [hunk 2]
```diff
@@ -46,18 +46,18 @@
...
+  --cpd-font-body-md-medium: var(--cpd-font-weight-medium) 0.938rem/var(--cpd-font-line-height-regular) var(--cpd-font-family-sans);
+  --cpd-font-body-md-regular: var(--cpd-font-weight-regular) 0.938rem/var(--cpd-font-line-height-regular) var(--cpd-font-family-sans);
+  --cpd-font-body-sm-semibold: var(--cpd-font-weight-semibold) 0.813rem/var(--cpd-font-line-height-regular) var(--cpd-font-family-sans);
+  --cpd-font-body-sm-medium: var(--cpd-font-weight-medium) 0.813rem/var(--cpd-font-line-height-regular) var(--cpd-font-family-sans);
+  --cpd-font-body-sm-regular: var(--cpd-font-weight-regular) 0.813rem/var(--cpd-font-line-height-regular) var(--cpd-font-family-sans);
+  --cpd-font-body-xs-semibold: var(--cpd-font-weight-semibold) 0.688rem/var(--cpd-font-line-height-regular) var(--cpd-font-family-sans);
+  --cpd-font-body-xs-medium: var(--cpd-font-weight-medium) 0.688rem/var(--cpd-font-line-height-regular) var(--cpd-font-family-sans);
+  --cpd-font-body-xs-regular: var(--cpd-font-weight-regular) 0.688rem/var(--cpd-font-line-height-regular) var(--cpd-font-family-sans);
```
Comment (sandhose) [inline assets/web/css/cpd-common-base.css:60]: This shouldn't change anything, the value is just (for some reason) inlined

File: assets/web/js/cpdDark.js [hunk 3]
```diff
@@ -56,18 +56,18 @@ export const cpdFontLetterSpacingHeadingSm = "-0.0166em";
...
+export const cpdFontBodyXsSemibold = "600 0.688rem/1.5 Inter";
+export const cpdFontBodySmRegular = "400 0.813rem/1.5 Inter";
+export const cpdFontBodySmMedium = "500 0.813rem/1.5 Inter";
+export const cpdFontBodySmSemibold = "600 0.813rem/1.5 Inter";
+export const cpdFontBodyMdRegular = "400 0.938rem/1.5 Inter";
+export const cpdFontBodyMdMedium = "500 0.938rem/1.5 Inter";
+export const cpdFontBodyMdSemibold = "600 0.938rem/1.5 Inter";
+export const cpdFontBodyLgRegular = "400 1.063rem/1.5 Inter";
+export const cpdFontBodyLgMedium = "500 1.063rem/1.5 Inter";
+export const cpdFontBodyLgSemibold = "600 1.063rem/1.5 Inter";
```
Comment (sandhose) [inline assets/web/js/cpdDark.js:70]: The rounding is a little bit different, I don't know how much this impacts stuff

File: package.json [hunk 4]
```diff
@@ -28,17 +29,17 @@
...
-    "typescript": "^4.9.5"
+    "lodash-es": "^4.17.21",
+    "style-dictionary": "3.8.0",
+    "svg2vectordrawable": "^2.9.1",
+    "tsx": "^4.15.6",
+    "typescript": "^5.4.5"
   },
-  "dependencies": {
-    "svg2vectordrawable": "^2.9.1"
-  }
```
Comment (sandhose) [inline package.json:43]: I was always a little bit annoyed that this was in my tree for some reason in MAS…

**Comment by robintown** [comment]
Uh oh, it looks like GitHub has closed this PR automatically now that I deleted the lint branch. That's one of the oddities that makes me hesitant to always set the base branch as the merge target when creating stacked PRs.

**Comment by sandhose** [comment]
@robintown No worries, I rebased and reopened in #89

**Pull request closed without merging**